### ISSUE TITLE
[Feat] Find contours algorithm

### ISSUE THREAD

**Issue by edgarriba** [kornia/kornia#939]
## 🚀 Feature
Implement in pytorch the find contours algorithm similar from OpenCV

Resources:
- https://docs.opencv.org/master/d3/dc0/group__imgproc__shape.html#gadf1ad6a0b82947fa1fe3c3d497f260e0

**Comment by MrParosk** [comment]
How is this algorithm intended to be implemented? After reading a description of it (e.g. https://theailearner.com/tag/suzuki-contour-algorithm-opencv/) the algorithm is fairly iterative and therefore might be slow in Python? Or shall it be written in C++ and included as a torchscript op? Or something else?

**Comment by edgarriba** [comment]
@MrParosk you are right - this is an open question how to handle algorithms with dedicated c++/cuda kernels that might require compiling/serving to specific torch/cuda versions. /cc @t-vi any thoughts on that ?

However, this might require a bit of more maturity from infra side, I would suggest to first try a naive python implementation to have as a baseline. There's a pending research work in order to find potential implementations to adopt, that later can be improved.

**Comment by MohamedAliRashad** [comment]
Have there been any update on this ?
A `findContour` like algorithm in kornia that fits nicely with PyTorch will be a great addon to the community.

**Comment by edgarriba** [comment]
@MohamedAliRashad as far as I know, no one is working on that feature yet. Contributions are welcomed

**Comment by manbehindthemadness** [comment]
I too have been searching for exactly this; however, after doing a code review with some of my people on: ``https://github.com/opencv/opencv/blob/4.x/modules/imgproc/src/contours.cpp`` It was decided that none of us has the platform experience to rewrite it into a maintainable enhancement. Sure would be nice though as I am not aware of a GPU/TPU enabled equivalent library anywhere.
